Automattic/wp-calypso
# Storage Add-ons: Expose add-on upsells to plans page

Related to https://github.com/Automattic/martech/issues/2023

## Proposed Changes

* Add storage add-on upsells to business and WooCommerce plans in the /plans page
* Add storage add-on upsell to spotlight card if business or WooCommerce plan
* Don't show add-on price if the storage add-on has already been purchased
* Don't show add-on prices if the user can no longer purchase storage add-ons given the 150GB max site storage limits

**Important:**
* Currently, the user can purchase a 50GB storage upgrade add-on for a total of 100GB site storage. They can, alternatively, purchase a 100GB storage upgrade for 150GB total site storage. Users cannot, however, purchase both, for a total of 200GB site storage. p1694555758071819/1694555365.713609-slack-CFFF01Q4V
* Storage add-on upsell UI is stale if revisiting the `/plans` page after recently purchasing a storage add-on. Normally the storage add-on dropdown will default to the most recently purchased plan. For a few minutes after a recent purchase though, it will not. This behavior is what we see in the `/add-ons` page as well.

## GIFs

![2023-10-24 10 29 58](https://github.com/Automattic/wp-calypso/assets/5414230/326d31f5-0c00-42c6-b8dd-b923411e1f44)

![2023-10-26 01 08 30](https://github.com/Automattic/wp-calypso/assets/5414230/a42907ba-7265-4660-a58d-d07f210305ce)

## Testing Instructions



* Navigate to /plans/{SITE_SLUG}
* Verify that storage add-on upsell dropdowns are displayed for business and WooCommerce plans
* Verify that, when a storage add-on upsell is selected, plan header prices and billing timeframes update accordingly
* Verify that storage add-ons upgrades are added to the checkout cart as expected
* Repeat the same verifications when the business or WooCommerce plan are in the spotlight card
* Verify that, when a storage add-on is purchased, it is rendered in the dropdown as the default on initial page load ( for recently purchased add-ons, stale data may be presented for a minute or two until updated. This same behavior is seen in the /add-ons page )
* Verify that, after a storage add-on has been purchased, selecting a lower tier storage add-on option disables the plan's CTA and asks the user to contact support

## Pre-merge Checklist
